docs/runtime/beam-metrics.md

```diff
@@ -0,0 +1,947 @@
+| `ets_alloc` | Allocator used for ets data. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `driver_alloc` | Allocator used for driver data. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `literal_alloc` | Allocator used for constant terms in Erlang code. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `sl_alloc` | Allocator used for memory blocks that are expected to be short-lived. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `ll_alloc` | Allocator used for memory blocks that are expected to be long-lived, for example, Erlang code. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `fix_alloc` | A fast allocator used for some frequently used fixed size data types. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `std_alloc` | Allocator used for most memory blocks not allocated through any of the other allocators described above. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `sys_alloc` | This is normally the default malloc implementation used on the specific OS. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `mseg_alloc` | A memory segment allocator. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+`beam.memory.allocators.kind` MUST be one of the following:
+
+| Value  | Description | Stability |
+|---|---|---|
+| `sbcs` | sbcs | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `mbcs` | mbcs | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `mbcs_pool` | mbcs_pool | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+<!-- endsemconv -->
+
+### Metric: `beam.memory.allocators.carrier.size`
+
+This metric is [recommended][MetricRecommended].
+
+<!-- semconv metric.beam.memory.allocators.carrier.size(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `beam.memory.allocators.carrier.size` | UpDownCounter | `By` | Total size of the memory carriers for the different allocators in the VM. See erts_alloc(3). | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+<!-- endsemconv -->
+
+<!-- semconv metric.beam.memory.allocators.carrier.size(full) -->
+| Attribute  | Type | Description  | Examples  | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) | Stability |
+|---|---|---|---|---|---|
+| `beam.memory.allocators.alloc` | string | The allocation type. | `sys_alloc` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `beam.memory.allocators.instance_no` | int | The logical instance number [0..n] | `1` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

Comment:
I feel like using `beam.memory.allocators.instance` here is not losing any descriptiveness but also lines up with what the OTP docs would provide?

Reply:
agreed.